### ISSUE TITLE
[Post Hackdays] edit_box spelunking

### ISSUE THREAD

**Issue by Schwad** [scarpe-team/scarpe#52]
This is the kind of work I think we'll do for a lot of our main widgets after hackdays. 

@DRBragg implemented edit_box [here](https://github.com/Schwad/scarpe/pull/47) which is looking good to me. It appears there might be a few different (non-idiomatic?) uses of edit_box in some examples so it might be worth a few cycles seeing where we can improve compatibility or decide what we don't want to support

## Context:

```ruby
Shoes.app do
  stack :margin => 10 do
    @edit = edit_box :width => 1.0 do
      @para.text = @edit.text
    end
    @para = para ""
  end
end
```

```ruby
Shoes.app do
  edit_box do |e|
    @counter.text = e.text.size
  end
  @counter = strong("0")
  para @counter, " characters"
end
```

```ruby
edit_box do |e|
  e.text = 'hiya'
end

edit_box "wowwwww"

@dayum = edit_box do 
  @other_thing = @dayum.text
end
````

**Comment by noahgibbs** [comment]
Ooh... Yeah, our edit box is wrong by Shoes standards right now. Here's a chunk from the manual:

```
To illustrate:

{{{
 Shoes.app do
   stack margin: 10 do
     @edit = edit_box width: 1.0 do
       @para.text = @edit.text
     end
     @para = para ""
   end
 end
}}}

This app will copy anything you paste into the edit box and display it in a
Shoes paragraph.  You can try copying some foreign text (such as Greek or
Japanese) into this box to see how it displays
```

We don't just need to change edit_box, we need to change any examples that use it in our current way. D'oh!

**Comment by noahgibbs** [comment]
The legacy examples seem to use that block as a changed callback, though.

**Comment by noahgibbs** [comment]
This should be fixed by PR #296.
